### ISSUE TITLE
Custom css variables not work for Seed Token

### ISSUE THREAD

**Issue by Rey-Wang** [ant-design/ant-design#49041]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/sandpack-project-forked-dd5cn9?file=%2Fvariable.css%3A17%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwbws35w00073p6p4nnvnbnz%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwbws35w00033p6p8qcs17dz%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwbws35w00043p6p2qdgn03e%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwbws35w00063p6pnv2kf58y%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwbws35w00033p6p8qcs17dz%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbws35v00023p6ptcc7ut4m%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fapp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A3%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A4%252C%2522endColumn%2522%253A30%257D%255D%257D%252C%257B%2522id%2522%253A%2522clwixmms700023p6p1lec2pwd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A17%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A17%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fvariable.css%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clwbws35w00033p6p8qcs17dz%2522%252C%2522activeTabId%2522%253A%2522clwixmms700023p6p1lec2pwd%2522%257D%252C%2522clwbws35w00063p6pnv2kf58y%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwbws35w00053p6phz7zigt5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clwbws35w00063p6pnv2kf58y%2522%252C%2522activeTabId%2522%253A%2522clwbws35w00053p6phz7zigt5%2522%257D%252C%2522clwbws35w00043p6p2qdgn03e%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clwbws35w00043p6p2qdgn03e%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

- set css variable for seed token and component token
- check if the style is applied or not

### What is expected?

seed token configuration worked

### What is actually happening?

failed to load seed token configuration

| Environment | Info |
| --- | --- |
| antd | 5.17.3 |
| React | 18 |
| System | macos 14.1 |
| Browser | chrome 125 |

**Comment by Rey-Wang** [comment]
Any updates on this?

**Comment by zhaokejian** [comment]
Same Problem!

**Comment by Berd74** [comment]
Same problem here!

**Comment by Berd74** [comment]
It looks like the antd doesn't support using CSS variables in tokens.
You can convert in JS the color behind the CSS variable to simple rgba or hex.

Here example with theme changing system.

index.css
```
html {
  --bg-base: #{lighten(lch(29.2345% 64.2 137), 20%)};
}
html.dark {
  --bg-base: #358;
}
```

App.tsx
```
function App() {

  const [isDark, setIsDark] = useState<boolean>(true); // add localStorage getter here to get selected theme
  const [number, setNumber] = useState<number>(1);

  const toggleTheme = () => {
    if (isDark) {
      document.documentElement.classList.remove('dark');
    } else {
      document.documentElement.classList.add('dark');
    }
    setIsDark(!isDark);
    // add localStorage setter here to save selected theme
  };

  return (
    <ConfigProvider
      theme={{
        cssVar: true,
        hashed: false,
        algorithm: isDark ? theme.darkAlgorithm : theme.defaultAlgorithm,
        token: {
          colorBgBase: cssVar('--bg-secondary'),
        },
        components: {
          Menu: MenuToken,
          Button: {}
        }
      }}
    >

      <main className='flex h-screen bg-base'>
        <Button onClick={toggleTheme}>real theme changer</Button>
        <Button type={"primary"} onClick={() => setNumber(x => ++x)}>refresh {number}</Button>
        {/*your code*/}
      </main>
    </ConfigProvider>

  );
}
```

For the cssVar I am using "color" library.
```
import Color from 'color';
function cssVar(variableName: string): string {
  const rootStyles: CSSStyleDeclaration = getComputedStyle(document.documentElement);
  const colorValue: string = rootStyles.getPropertyValue(variableName).trim();
  return Color(colorValue).toString()
}
```

In your example in one case the color worked it is because the variable for the component is on the lowest layer of [Life of Design Token](https://ant.design/docs/react/customize-theme#life-of-design-token)

In short Antd is able to apply the CSS variable to the component directly but is not able to use it as a token. Based on the Seed Token the algorithm inside antd will automatically calculate and apply a series of corresponding colors according to the it. This is where antd fails - the algorithm doesn't understand the CSS varaible.

**Comment by ArturBaybulatov** [comment]
Related issue regarding `oklch` notation support in antd:

https://github.com/ant-design/ant-design/issues/48698

**Comment by roddc** [comment]
为什么有些可以写变量有些就不可以呢？ colorBgContainer: 'var(--color-bg-container)' 这个就可以，但 colorPrimary: 'var(--color-bg-container)' 这个就不可以。 版本是5.29.1
